pilot/pkg/bootstrap/server.go

```diff
@@ -248,6 +250,12 @@ func (s *Server) initDiscoveryService(args *PilotArgs) error {
 		s.kubeRegistry.XDSUpdater = s.EnvoyXdsServer
 	}
 
+	// TODO: Split initDiscoveryService method in to createDiscoveryServer and initDiscoveryService so that, this special
+	// handling is not needed. Because of dependency ordering problem, we need to set this explicitly here.
+	if s.serviceEntryStore != nil {
```

Comment:
Once this PR is merged, I will do a followup and split this function in to multiple like initEnv, createDiscoveryServer and initDiscoveryService - and clean this and kuberegistry specific handling above.